pkg/manager/impl/validation/validation.go

```diff
@@ -268,3 +277,38 @@ func ValidateOutputData(outputData *core.LiteralMap, maxSizeInBytes int64) error
 	}
 	return errors.NewFlyteAdminErrorf(codes.ResourceExhausted, "Output data size exceeds platform configured threshold (%+v > %v)", outputSizeInBytes, maxSizeInBytes)
 }
+
+func ValidateDatetime(literal *core.Literal) error {
+	if literal == nil {
+		return nil
+	}
+
+	timestamp := literal.Value.(*core.Literal_Scalar).Scalar.Value.(*core.Scalar_Primitive).Primitive.GetValue().(*core.Primitive_Datetime).Datetime
+
+	err := timestamp.CheckValid()
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.InvalidArgument, err.Error())
+	}
+	return nil
+}
+
+func isDateTime(input *core.Literal) bool {
+	literalValue := input.Value
+	if reflect.ValueOf(literalValue).String() != "<*core.Literal_Scalar Value>" {
```

Comment:
instead of using reflection can you just check if the oneof value is nil, something like

`return input.GetScalar() != nil && input.GetScalar().GetPrimitive() != nil && inpuGetScalar().GetPrimitive().GetDatetime() != nil`

Reply:
Done, thanks for the hint!